sacred/config/custom_containers.py

```diff
@@ -185,10 +184,12 @@ class ReadOnlyDict(ReadOnlyContainer, dict):
     __setitem__ = ReadOnlyContainer._readonly
     __delitem__ = ReadOnlyContainer._readonly
 
-    def __init__(self, *args, message=None, **kwargs):
-        if message is None:
-            message = "This ReadOnlyDict is read-only!"
-        super().__init__(*args, message=message, **kwargs)
+    def __init__(self, d, message=None):
+        ReadOnlyContainer.__init__(self, message)
+        dict.__init__(self, d)
+
+    def __reduce__(self):
+        return ReadOnlyDict, (dict(self), self.message)
```

Comment:
This _could_ be moved to the superclass, but I don't know if this actually makes things simpler:

```python
def __reduce__(self):
    return self.__class__, (self.__copy__(), self.message)
```

Reply:
Suggestion taken. It looks about the same and is fewer lines of code